app/javascript/app/ssn-field.js

```diff
@@ -0,0 +1,47 @@
+import Cleave from 'cleave.js';
+
+const { I18n } = window.LoginGov;
+
+/* eslint-disable no-new */
+function formatSSNField() {
+  const inputs = document.querySelectorAll('input.ssn-toggle[type="password"]');
+
+  if (inputs) {
+    [].slice.call(inputs).forEach((input, i) => {
+      const el = `
+        <div class="mt1 right">
+          <label class="btn-border" for="ssn-toggle-${i}">
+            <div class="checkbox">
+              <input id="ssn-toggle-${i}" type="checkbox">
+              <span class="indicator"></span>
+              ${I18n.t('forms.ssn.show')}
+            </div>
+          </label>
+        </div>`;
+      input.insertAdjacentHTML('afterend', el);
+
+      let ssnCleave;
+      const toggle = document.getElementById(`ssn-toggle-${i}`);
+      toggle.addEventListener('change', function () {
+        input.type = toggle.checked ? 'text' : 'password';
+
+        if (!toggle.checked) {
+          if (ssnCleave) {
+            ssnCleave.destroy();
+          }
+
+          input.value = input.value.replace(/-/g, '');
+        } else {
+          ssnCleave.destroy();
```

Comment:
If we want to destroy the current instance regardless of condition, we could pull it out as common. Could also use [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) to avoid checking if it exists.

```suggestion

        ssnCleave?.destroy();

        if (!toggle.checked) {
          input.value = input.value.replace(/-/g, '');
        } else {
```